modules/ROOT/pages/ftbfs.adoc

```diff
@@ -1,31 +1,123 @@
-= Handling ELN build failures =
+= Resolving ELN build failures =
 
-[NOTE]
-====
-This is currently a rough placeholder.
-Additional details will be provided soon.
-====
+This page provides information to assist package maintainers remediate
+ELN build failures.
 
-This page provides information to assist package maintainers remediate ELN build failures.
 
 == Background
 
+ELN is a buildroot and compose process for Fedora that takes Fedora Rawhide
+https://src.fedoraproject.org/[dist-git] sources and emulates a Red Hat
+Enterprise Linux compose.
+
+ELN sources come from the Rawhide (master) branch in dist-git.
+There is _not_ a separate branch for ELN.
+
 Note what’s different with the xref:buildroot.adoc[buildroot] for ELN,
-as well as details about xref:buildroot.adoc#building[how to build] using the ELN buildroot.
+as well as details about xref:buildroot.adoc#building[how to build] using
+the ELN buildroot.
+
+The set of packages that currently need to build for ELN can be found at
+https://tiny.distro.builders/view\--view-eln.html.
+The expanded package set including build dependencies can be found at
+https://tiny.distro.builders/view\--view-eln-and-buildroot.html.
 
-A list of packages that currently need to build for ELN can be found at
-https://tiny.distro.builders/view\--view-eln.html
 
 == Fixing ELN build failures
 
-=== Guidelines
-To be provided.
+=== If the package doesn't build for Rawhide, fix that first!
+
+IMPORTANT: If the package doesn't build for Rawhide, fix that first!
+
+If the package doesn't build for Rawhide, fix that first!
+
+Is that message load and clear enough?
+Getting the package build fixed for Rawhide may very well be all that's
+needed to fix it for ELN as well.
+
+
+=== Accounting for differences between Fedora and RHEL
+
+Most build failures can be fixed by modifying the SPEC file.
+However, such modifications should be kept to a minimum.
+
+If your package needs to institute different code paths or package
+dependencies between Fedora and RHEL, it is important to only account
+for the difference for the releases where it is required.
+
+Generally, this means limiting the difference in old versions of Fedora,
+while letting the next version of RHEL adopt the Fedora code path.
+
+As one common scenario where a difference is required, RHEL often ends
+up with a smaller set of dependencies than Fedora.
+The `%\{fedora}` and `%\{rhel}` macros, when defined, contain the major
+release versions per distro.
+
+Consider this SPEC file snippet:
+
+```
+%if 0%{?fedora} >= 29 || 0%{?rhel} >= 9
+BuildRequires: SomePackage
+%endif
+```
+
+This conditional will be met when `%\{rhel}` is defined as 9 or greater.
+It is also shareable between earlier and later versions of Fedora and RHEL.
+Even better is using `<` to limit behaviors to RHEL 8 and earlier:
```

Comment:
I don't think we should globally say it's better.  It depends on the circumstances.  What if we changed it from
  Even better is using
to
  Often it is better to use

Reply:
I agree. I just pushed another commit incorporating that, as well as adding a short section about handling the recurring python 2 deprecation failures. (Although most of those have probably already been fixed.)

Reply:
Update looks good.
I like the call out to Python2.